src/hsm.cpp

```diff
@@ -261,8 +261,9 @@ openssl::SSL_EVP_PKEY_Ptr HsmEngine::generateKey(const ECCSpec &spec,
     pkcs11ECCSpec.curve = curve.c_str();
 
     PKCS11_params _params;
-    _params.extractable = static_cast<unsigned char>(params.cka_extractable);
-    _params.sensitive = static_cast<unsigned char>(params.cka_sensitive);
+    // If the key is extractable it shouldn't be sensitive and vice versa
```

Comment:
I guess this makes sense in principle but is there a strong reason why do we want to enforce it?

Reply:
@istepic This was requested in the review from @garw in #155. We should hide the PKCS#11 internals...